mw-webapp/src/logic/wayPage/reportsTable/WayColumns.tsx

```diff
@@ -383,30 +382,36 @@ export const Columns = (props: ColumnsProps) => {
           /**
            * TODO: need to delete this check after we will add possibility to add comments only for mentors
            */
-          if (!mentor) {
-            return "User is not a mentor";
+          if (!user?.displayName) {
+            return "User is not registered";
           }
-          const userName = mentor.name;
+          const userName = mentor ? mentor.name : user.displayName;
 
           return userName;
         };
 
         return (
+          user &&
           <TableCell
-            buttonValue="add comment"
-            onButtonClick={() => createComment(user?.uid)}
+            buttonValue={user?.uid === ownerUuid || Array.from(props.mentors.keys()).includes(user?.uid)
+              ? "add comment"
+              : ""
+            }
+            onButtonClick={() => createComment(user.uid)}
           >
             {row.original.comments
               .map((comment) => (
                 <CellItem key={comment.uuid}>
```

Comment:
CellItem is not a cellItem, looks like it's kind of horizontal container. Refactoring required